javagen/src/main/java/com/azure/autorest/mapper/ClientMapper.java

```diff
@@ -343,6 +318,102 @@ public Client map(CodeModel codeModel) {
+                                    convenienceExamples.add(convenienceExample);
+                                    convenienceExampleNameSet.add(filename);
+                                }
+                            }
+                        }));
+            }
+        };
+
+        // convenience examples
+        syncClients.stream().filter(c -> !CoreUtils.isNullOrEmpty(c.getConvenienceMethods()))
+                .forEach(c -> c.getConvenienceMethods()
+                        .forEach(m -> handleConvenienceExample.accept(c, m)));
+        builder.convenienceExamples(convenienceExamples);
+    }
+
+    private void addProtocolExamples(Client.Builder builder, List<AsyncSyncClient> syncClients) {
+        List<ProtocolExample> protocolExamples = new ArrayList<>();
+        Set<String> protocolExampleNameSet = new HashSet<>();
+
+        BiConsumer<AsyncSyncClient, ClientMethod> handleExample = (c, m) -> {
+            if (m.getMethodVisibility() == JavaVisibility.Public
+                    && !m.isImplementationOnly() &&
+                    (m.getType() == ClientMethodType.SimpleSyncRestResponse
+                    || m.getType() == ClientMethodType.PagingSync
+                    || m.getType() == ClientMethodType.LongRunningBeginSync)) {
+                ClientBuilder clientBuilder = c.getClientBuilder();
+                if (clientBuilder != null && m.getProxyMethod().getExamples() != null) {
+                    m.getProxyMethod().getExamples().forEach((name, example) -> {
+                        String filename = CodeNamer.toPascalCase(CodeNamer.removeInvalidCharacters(name));
+                        if (!protocolExampleNameSet.contains(filename)) {
+                            ProtocolExample protocolExample = new ProtocolExample(m, c, clientBuilder, filename, example);
+                            protocolExamples.add(protocolExample);
+                            protocolExampleNameSet.add(filename);
+                        }
+                    });
+                }
+            }
+        };
+
+        // protocol examples, exclude those that have convenience methods
+        syncClients.stream().filter(c -> c.getServiceClient() != null)
+                .forEach(c -> {
+                    Set<String> convenienceProxyMethodNames = new HashSet<>();
+                    if (c.getConvenienceMethods() != null) {
+                        convenienceProxyMethodNames.addAll(c.getConvenienceMethods().stream()
+                                .map(convenienceMethod -> convenienceMethod
+                                        .getProtocolMethod().getProxyMethod().getBaseName())
+                                .collect(Collectors.toSet()));
+                    }
+                    c.getServiceClient().getClientMethods()
+                            .stream()
+                            .filter(m -> !convenienceProxyMethodNames.contains(m.getProxyMethod().getBaseName()))
```

Comment:
only generate protocol examples for protocol methods without convenience methods